### DISCUSSION TITLE
`NMSFreeCoder` of `DETR3D`: `cls_scores.sigmoid()` then `cls_scores.view(-1).topk(max_num)`

### DISCUSSION THREAD

**Discussion by Ricardo-Yu** [open-mmlab/mmdetection3d#2980] [Q&A]
Hi, I'm confuesd about the `NMSFreeCoder` of `DETR3D`, which use `cls_scores.sigmoid()` then `cls_scores.view(-1).topk(max_num)` to generate final prediction bbox. Then when select `topK`, how to avoid won't select the same query multiple times? e.g. if `cls_scores.shape=[300, 10]` and `max_num=100`, then after `sigmoid()`, it's possible that an object_query have multiple high cls_scores since it's using sigmoid instead of softmax. After `.view(-1)` to flat as [3000], when apply `topK`, it's possible to select an object_query twice if it has 2 high cls scores from `sigmoid`? 

If an object_query may selected twice from bbox coder, then the DETR3D will predict 2 bbox3d fully overlapped, but with different classes? Will this be an issue when using in autonomous driving tasks, which may need mutually exclusive classification for downstream to react accordingly?

https://github.com/open-mmlab/mmdetection3d/blob/fe25f7a51d36e3702f961e198894580d83c4387b/projects/DETR3D/detr3d/nms_free_coder.py#L55-L56

**Comment by nnop** [comment]
Have you figured out this question?
